integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala

```diff
@@ -124,15 +127,30 @@ object CarbonSession {
       getOrCreateCarbonSession(null, null)
     }
 
+    def getOrCreateCarbonSessionWithOutHive(): SparkSession = {
```

Comment:
instead of providing this, can you add `def enableInMemCatalog(): CarbonBuilder`, so that user can do:
```
    SparkSession.builder
        .enableInMemCatalog
        .getOrCreateCarbonSession(x,x)
```